chapter2/agent-skills-ppt/run_official_experiment.py

```
@@ -96,9 +60,84 @@ def main() -> int:
+    fallback = Path.home() / ".kimi-code" / "bin" / "kimi"
+    if fallback.is_file():
+        return str(fallback)
+    raise RuntimeError("Kimi Code CLI not found on PATH or at ~/.kimi-code/bin/kimi")
+
+
+def stream_process(command: list[str], workspace: Path, env: dict, stream_path: Path,
+                   stderr_path: Path, tag: str) -> int:
+    with stream_path.open("w", encoding="utf-8") as stdout_file, stderr_path.open(
+        "w", encoding="utf-8"
+    ) as stderr_file:
+        process = subprocess.Popen(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)